### PR TITLE
Feature/CCM 5366 Enable Cloudwatch Alarms

### PR DESCRIPTION
**Pull request by jamesthompson26-nhs**
## Description

Add basic error logging and alarms for the Step Function component of the Reporting MVP.

## Context

To determine the cause of issues we need a way of alerting on failed/timed-out/aborted jobs.

## Type of changes

Adds a cloudwatch log group.
Adds 3 cloudwatch alarms. 
Configures the step function to write to the log group and modifies its IAM policy to be able to do so.

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: infrastructure/terraform/components/reporting/files/state.tmpl.json
Comment (jamesthompson26-nhs) [inline infrastructure/terraform/components/reporting/files/state.tmpl.json:1]: @sidnhs - VSCode fixed the indentation automatically here when I was rebasing!

**Review by sidnhs — APPROVED** [review]

**Review by aidenvaines-bjss — COMMENTED** [review]
There doesnt seem to be a notification sent out for any of these alarms?

File: infrastructure/terraform/components/reporting/sfn_state_machine_athena.tf [hunk 1]
```diff
@@ -169,4 +175,25 @@ data "aws_iam_policy_document" "sfn_athena" {
...
+      "logs:DeleteLogDelivery",
+      "logs:ListLogDeliveries",
+      "logs:PutLogEvents",
+      "logs:PutResourcePolicy",
+      "logs:DescribeResourcePolicies",
+      "logs:DescribeLogGroups"
+    ]
+
+    resources = [
+      "*", ## https://docs.aws.amazon.com/step-functions/latest/dg/cw-logs.html
```
Comment (aidenvaines-bjss) [inline infrastructure/terraform/components/reporting/sfn_state_machine_athena.tf:196]: Can this be scoped better please?

File: infrastructure/terraform/components/reporting/variables.tf [hunk 2]
```diff
@@ -79,3 +79,9 @@ variable "core_env" {
   description = "The core environment that contains the corresponding Glue table/S3 buckets etc."
   default     = "internal-dev"
 }
+
+variable "log_retention_days" {
+  type        = number
+  description = "How many days to retain the logs generated by the step function"
+  default     = 60
```
Comment (aidenvaines-bjss) [inline infrastructure/terraform/components/reporting/variables.tf:86]: im sure ive seen the default for CW logs be 30 elsewhere in this repo due to Splunk forwarding, is this ok @sidnhs

**Review by aidenvaines-bjss — DISMISSED** [review]
Doesnt seem to be any targets for CW Alarms, whats the point in them juse going red?

File: infrastructure/terraform/components/reporting/sfn_state_machine_athena.tf (hunk 1 above)
Comment (jamesthompson26-nhs) [reply infrastructure/terraform/components/reporting/sfn_state_machine_athena.tf:196]: The link in the comment explains it better than I can. I did initially have it scoped to the log group itself but it doesn't work.

File: infrastructure/terraform/components/reporting/variables.tf (hunk 2 above)
Comment (jamesthompson26-nhs) [reply infrastructure/terraform/components/reporting/variables.tf:86]: it looked to be set to 60 in most cases across the comms-mgr repo so I went with that.

Comment (sidnhs) [reply infrastructure/terraform/components/reporting/variables.tf:86]: iirc the lambda logs are 30 and api gw logs are 60 by default on core

Comment (jamesthompson26-nhs) [reply infrastructure/terraform/components/reporting/variables.tf:86]: ah ok, I'll amend to 30 to be in line with the lambda logs as they share a lot in common.

File: infrastructure/terraform/components/reporting/sfn_state_machine_athena.tf (hunk 1 above)
Comment (jamesthompson26-nhs) [reply infrastructure/terraform/components/reporting/sfn_state_machine_athena.tf]: I've addressed this to some extent by splitting out into two separate statements, scoping where I can and adding a conditional to ensure the requestingtag is for the same environment.

**Review by aidenvaines-bjss — APPROVED** [review]

**Review by sidnhs — APPROVED** [review]

**Pull request merged**